### PR TITLE
[Broker] Adjust the validation for policy schemaCompatibilityStrategy

### PR DESCRIPTION
**Pull request by yuruguo**
### Motivation
In #13297, We should not add `GET_SCHEMA_COMPATIBILITY_STRATEGY` and `SET_SCHEMA_COMPATIBILITY_STRATEGY` topic operation and complete the permission check on policy schemaCompatibilityStrategy by method `validateTopicOperationAsync()`, because schemaCompatibilityStrategy is a topic policy not a topic operation, we should use method `validateTopicPolicyOperation()` to complete.

### Documentation
- [x] `no-need-doc`

### CODE REVIEW

**Comment by yuruguo** [comment]
@codelipenghui @Jason918 @congbobo184 ptal

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/AdminResource.java [hunk 1]
```diff
@@ -766,39 +768,37 @@ protected void resumeAsyncResponseExceptionally(AsyncResponse asyncResponse, Thr
...
-        return validateTopicOperationAsync(topicName, TopicOperation.GET_SCHEMA_COMPATIBILITY_STRATEGY)
-                .thenCompose((__) -> {
-                    CompletableFuture<SchemaCompatibilityStrategy> future;
-                    if (config().isTopicLevelPoliciesEnabled()) {
-                        future = getTopicPoliciesAsyncWithRetry(topicName)
-                                .thenApply(op -> op.map(TopicPolicies::getSchemaCompatibilityStrategy).orElse(null));
-                    } else {
-                        future = CompletableFuture.completedFuture(null);
-                    }
+        validateTopicPolicyOperation(topicName, PolicyName.SCHEMA_COMPATIBILITY_STRATEGY, PolicyOperation.READ);
```
Comment (Jason918) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/AdminResource.java:771]: This is blocking method. Please use `validateTopicPolicyOperationAsync`.

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java [hunk 2]
```diff
@@ -4826,25 +4826,25 @@ private void internalGetReplicatedSubscriptionStatusForNonPartitionedTopic(Async
...
         }
-        return validateTopicOperationAsync(topicName, TopicOperation.GET_SCHEMA_COMPATIBILITY_STRATEGY)
-                .thenCompose(n -> getTopicPoliciesAsyncWithRetry(topicName).thenApply(op -> {
-                    if (!op.isPresent()) {
-                        return null;
-                    }
-                    SchemaCompatibilityStrategy strategy = op.get().getSchemaCompatibilityStrategy();
-                    return SchemaCompatibilityStrategy.isUndefined(strategy) ? null : strategy;
-                }));
+        validateTopicPolicyOperation(topicName, PolicyName.SCHEMA_COMPATIBILITY_STRATEGY, PolicyOperation.READ);
```
Comment (Jason918) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java:4829]: Same here.

**Review by Jason918 — CHANGES_REQUESTED** [review]

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/AdminResource.java (hunk 1 above)
Comment (yuruguo) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/AdminResource.java:771]: I found that pulsar does not provide async method `validateTopicPolicyOperationAsync`, maybe we should add this method and replace all synchronous method `validateTopicPolicyOperation`.

Comment (Jason918) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/AdminResource.java:771]: https://github.com/apache/pulsar/blob/a0fe8562694013b11910b39ebf4dc703c1400bf4/pulsar-broker/src/main/java/org/apache/pulsar/broker/web/PulsarWebResource.java#L1101 
This one?

**Review by Jason918 — APPROVED** [review]
LGTM

**Comment by yuruguo** [comment]
@eolivelli @codelipenghui PTAL when you have a moment

**Review by nodece — APPROVED** [review]
LGTM

**Review by eolivelli — APPROVED** [review]
Lgtm

CN you please remember me the PIP in which we are changing all the permissions?

**Comment by nodece** [comment]
> Lgtm
> 
> CN you please remember me the PIP in which we are changing all the permissions?

@eolivelli Sorry, I should make a PIP for #13297. Do we need to make PIR for these changes?

**Comment by yuruguo** [comment]
@codelipenghui @congbobo184 PTAL

**Comment by eolivelli** [comment]
I mean that during the past months I have seen many patches around permission validation.
I wonder if there is a general umbrella issue or some PIP.
This kind of changes must be documented carefully in the release notes,  ecause they have a big impact on security

**Comment by gaoran10** [comment]
Could we add a test for this?

**Review by codelipenghui — APPROVED** [review]

**Comment by codelipenghui** [comment]
@yuruguo The change looks good to me, could you please help add a test?

**Comment by codelipenghui** [comment]
@eolivelli We don't have a release that contains #13297, so it will not have a big impact on security. The issue is #13297 implementation does not follow the current topic policy permission handling way.

> I wonder if there is a general umbrella issue or some PIP.

😭, looks this is a feature https://github.com/apache/pulsar/pull/6428 that was added almost 2 years ago, and no PIP there.

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java [hunk 3]
```diff
@@ -4837,7 +4839,9 @@ private void internalGetReplicatedSubscriptionStatusForNonPartitionedTopic(Async
     }
 
     protected CompletableFuture<Void> internalSetSchemaCompatibilityStrategy(SchemaCompatibilityStrategy strategy) {
-        return validateTopicOperationAsync(topicName, TopicOperation.SET_SCHEMA_COMPATIBILITY_STRATEGY)
+        return validateTopicPolicyOperationAsync(topicName,
+                PolicyName.SCHEMA_COMPATIBILITY_STRATEGY,
```
Comment (315157973) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java:4843]: Could there be a compatibility issue here? 
In addition, please add a unit test

Comment (codelipenghui) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java:4843]: No, https://github.com/apache/pulsar/pull/13297 only in the master

Comment (yuruguo) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java:4843]: > In addition, please add a unit test
Done

**Comment by yuruguo** [comment]
@codelipenghui @315157973 @gaoran10 The test has added and PTAL again, thx!

**Review by codelipenghui — APPROVED** [review]

**Comment by codelipenghui** [comment]
After checking the current implementation, looks like the PolicyName and PolicyOperation are only for the exception messages, only the tenant admin can access the topic policies.

**Review by congbobo184 — APPROVED** [review]

**Pull request merged**